app/src/main/java/com/zulip/android/models/Message.java

```diff
@@ -104,7 +105,7 @@
     private String subject;
     @SerializedName("timestamp")
     @DatabaseField(columnName = TIMESTAMP_FIELD)
-    private Date timestamp;
+    private long timestamp;
```

Comment:
@vishwesh3 I really appreciate how well you explained your point :) Although I still feel that changing the type of timestamp for a message is unnecessary. Ormlite [where.between](http://ormlite.com/javadoc/ormlite-core/com/j256/ormlite/stmt/Where.html#between(java.lang.String,%20java.lang.Object,%20java.lang.Object)) can take any type of data (in this case Date()) to return a modulated where. So I tried these [changes](https://gist.github.com/Sam1301/90a7271024e2b560d46e7f885a332198) to NarrowFilterByDate.java and it worked for me. The code in my example could obviously be more concise but I guess you get the point :)